packages/phone-number-privacy/common/src/utils/input-validation.ts

```diff
@@ -1,28 +1,39 @@
 import { isValidAddress, trimLeading0x } from '@celo/utils/lib/address'
+import isBase64 from 'is-base64'
+import { isValidPhoneNumber } from 'libphonenumber-js'
+import {
+  GetBlindedMessageSigRequest,
+  GetContactMatchesRequest,
+  GetQuotaRequest,
+} from '../interfaces'
 import { REASONABLE_BODY_CHAR_LIMIT, REQUEST_EXPIRY_WINDOW_MS } from './constants'
 
 export function hasValidAccountParam(requestBody: any): boolean {
   return requestBody.account && isValidAddress(requestBody.account)
 }
 
-export function hasValidUserPhoneNumberParam(requestBody: any): boolean {
-  return !!requestBody.userPhoneNumber
+export function hasValidUserPhoneNumberParam(requestBody: GetContactMatchesRequest): boolean {
+  return !!requestBody.userPhoneNumber && isValidPhoneNumber(requestBody.userPhoneNumber)
 }
 
-export function hasValidContractPhoneNumbersParam(requestBody: any): boolean {
-  return requestBody.contactPhoneNumbers && Array.isArray(requestBody.contactPhoneNumbers)
+export function hasValidContactPhoneNumbersParam(requestBody: GetContactMatchesRequest): boolean {
+  return (
+    !!requestBody.contactPhoneNumbers &&
+    Array.isArray(requestBody.contactPhoneNumbers) &&
```

Comment:
Also, we should probably add a check that the contactPhoneNumbers array is non-empty. It looks like we divide by the length of the array here https://github.com/celo-org/celo-monorepo/blob/723028ea170e459da762ecc4ca9e6309f4005fde/packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts#L63. It might be worth adding a test case for this as well.